tests/Unit/PdoConnectionTest.php

```diff
@@ -153,8 +158,60 @@ public function testRollBack()
      */
     public function testCommit()
     {
+        // begin and commit transaction
         $this->connection->setConnection($this->mockConnection('commit', true));
+        $this->connection->begin();
         $this->assertTrue($this->connection->commit());
+
+        // Since there is no open transaction, this should return false
+        $this->assertFalse($this->connection->commit());
+    }
+
+    public function nestedTransactionData()
+    {
+        return array(
+            array(array('commit', 'commit', 'commit'), true),
+            array(array('rollback', 'rollback', 'rollback'), true),
+            array(array('commit', 'commit', 'rollback'), true),
+            array(array('rollback', 'commit', 'rollback'), true),
+            array(array('rollback', 'commit'), false),
+            array(array('commit', 'rollback', 'commit'), false)
+        );
+    }
+
+    /**
+     * @covers ::__construct
+     * @covers ::begin
+     * @covers ::commit
+     * @covers ::rollback
+     * @covers ::connect
+     * @covers ::getConnection
+     * @covers ::setConnection
+     * @dataProvider nestedTransactionData
+     *
+     * @param array $actions
+     * @param bool $return
+     */
+    public function testNestedTransactions(array $actions, $return)
+    {
+        $this->connection->setConnection($this->mockConnection('beginTransaction', true));
+        foreach ($actions as $action) {
+            $this->assertTrue($this->connection->begin());
+        }
+
+        end($actions);
+        $end_index = key($actions);
+        reset($actions);
```

Comment:
does PSR2 allow underscore in variable names? I thought they had to be `camelCase`...